lib/swm050/gpio.c

```diff
@@ -187,11 +185,123 @@ is enabled, GPIO and AF of the SWD pins will be both unavailable.
 */
 void gpio_sel_swd(bool en)
 {
-	if (en) {
-		SWD_SEL = 1;
-	} else {
-		SWD_SEL = 0;
-	}
+	SWD_SEL = en;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets the pins as external interrupts, rather than normal GPIO
+
+Enable interrupts on the selected pins.  If you want to quickly
+switch on and off interrupts, use gpio_int_mask() after calling this.
+
+@param[in] gpios Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+*/
+void gpio_int_en(uint16_t gpios)
```

Comment:
I _really_ want this to be gpio_irq_enable or gpio_int_enable if you must.  Also, there's no method to ever clear these bits.  I know that you say that using the mask function is "faster" but there's still no way to turn them off.